src/test/java/org/jsoup/nodes/DocumentTest.java

```diff
@@ -157,4 +162,229 @@
         assertEquals(docA.hashCode(), docB.hashCode());
         assertFalse(docA.hashCode() == docC.hashCode());
     }
+    
+    @Test
+    public void testMetaCharsetUpdateUtf8() {
+        final Document doc = createHtmlDocument("changeThis");
+        doc.updateMetaCharset(true);
+        doc.charset(Charset.forName(charsetUtf8));
+        
+        final String htmlCharsetUTF8 = "<html>\n" +
+                                        " <head>\n" +
+                                        "  <meta charset=\"" + charsetUtf8 + "\">\n" +
+                                        " </head>\n" +
+                                        " <body></body>\n" +
+                                        "</html>";
+        assertEquals(htmlCharsetUTF8, doc.toString());
+        
+        Element selectedElement = doc.select("meta[charset]").first();
+        assertEquals(charsetUtf8, doc.charset().displayName());
```

Comment:
I believe every instance of displayName() should be name(). Latter is canonical, but former will / can vary depending on Locale. http://docs.oracle.com/javase/7/docs/api/java/nio/charset/Charset.html#displayName()